internal/compiler/module/resolver.go

```diff
@@ -145,6 +145,28 @@ func (r *Resolver) GetPackageScopeForPath(directory string) *packagejson.InfoCac
 	return (&resolutionState{compilerOptions: r.compilerOptions, resolver: r}).getPackageScopeForPath(directory)
 }
 
+func (r *Resolver) GetPackageJsonTypeIfApplicable(path string) string {
+	if tspath.FileExtensionIsOneOf(path, []string{".mts", ".cts", "mjs", "cjs"}) {
```

Comment:
Some of these have dots and some don’t, so I don’t think it’s going to work 😁 IIRC we have constants `tspath.ExtensionMts` and so on.